Tests: distributor_test.go fails when the system has no interface name in [eth0, en0, lo0]

**Describe the bug**
Distributor tests fail when the test system does not have any network interfaces named eth0, en0, or lo0.

**To Reproduce**
`make test` on a system without any interfaces named eth0, en0, or lo0.

**Expected behavior**
`make test` should pass.

**Environment:**
 - Debian buster
 - Golang 1.13.6-2~bpo10+1

**Screenshots, Promtail config, or terminal output**
```
--- FAIL: TestDistributor_PushIngestionRateLimiter (0.00s)
    --- PASS: TestDistributor_PushIngestionRateLimiter/local_strategy:_limit_should_be_set_to_each_distributor (0.00s)
    --- FAIL: TestDistributor_PushIngestionRateLimiter/global_strategy:_limit_should_be_evenly_shared_across_distributors (0.00s)
        distributor_test.go:230: 
            	Error Trace:	distributor_test.go:230
            	            				distributor_test.go:167
            	Error:      	Received unexpected error:
            	            	No address found for [eth0 en0 lo0]
            	Test:       	TestDistributor_PushIngestionRateLimiter/global_strategy:_limit_should_be_evenly_shared_across_distributors
    --- FAIL: TestDistributor_PushIngestionRateLimiter/global_strategy:_burst_should_set_to_each_distributor (0.00s)
        distributor_test.go:230: 
            	Error Trace:	distributor_test.go:230
            	            				distributor_test.go:167
            	Error:      	Received unexpected error:
            	            	No address found for [eth0 en0 lo0]
            	Test:       	TestDistributor_PushIngestionRateLimiter/global_strategy:_burst_should_set_to_each_distributor
```

--- a/pkg/distributor/distributor_test.go
+++ b/pkg/distributor/distributor_test.go
@@ -4,6 +4,7 @@ import (
 	"context"
 	"fmt"
 	"math/rand"
+	"net"
 	"net/http"
 	"strconv"
 	"strings"
@@ -193,6 +194,23 @@ func TestDistributor_PushIngestionRateLimiter(t *testing.T) {
 	}
 }
 
+// loopbackInterfaceName search for the name of a loopback interface in the list
+// of the system's network interfaces.
+func loopbackInterfaceName() (string, error) {
+	is, err := net.Interfaces()
+	if err != nil {
+		return "", fmt.Errorf("can't retrieve loopback interface name: %s", err)
+	}
+
+	for _, i := range is {
+		if i.Flags&net.FlagLoopback != 0 {
+			return i.Name, nil
+		}
+	}
+
+	return "", fmt.Errorf("can't retrieve loopback interface name")
+}
+
 func prepare(t *testing.T, limits *validation.Limits, kvStore kv.Client) *Distributor {
 	var (
 		distributorConfig Config
@@ -218,10 +236,13 @@ func prepare(t *testing.T, limits *validation.Limits, kvStore kv.Client) *Distri
 		})
 	}
 
+	loopbackName, err := loopbackInterfaceName()
+	require.NoError(t, err)
+
 	distributorConfig.DistributorRing.HeartbeatPeriod = 100 * time.Millisecond
 	distributorConfig.DistributorRing.InstanceID = strconv.Itoa(rand.Int())
 	distributorConfig.DistributorRing.KVStore.Mock = kvStore
-	distributorConfig.DistributorRing.InstanceInterfaceNames = []string{"eth0", "en0", "lo0"}
+	distributorConfig.DistributorRing.InstanceInterfaceNames = []string{loopbackName}
 	distributorConfig.factory = func(addr string) (ring_client.PoolClient, error) {
 		return ingesters[addr], nil
 	}

